.github/workflows/check-notebooks.yml

```diff
@@ -0,0 +1,37 @@
+name: "Check notebooks"
+on:
+  push:
+    branches:
+      - 'master'
+  pull_request:
+
+jobs:
+  check-notebooks:
+    runs-on: ubuntu-latest-8-cores
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          submodules: true
+
+      - name: Build container
+        run: docker-compose pull && docker-compose build && docker-compose up -d
```

Comment:
1. Suggest to make it one command (follow-up better option)
```shell
docker compose up --build --pull always -d
```
2. Suggest to move the exclude list into the notebooks folder (because of the next item)
3. Suggest to move the notebook test as an entry inside the docker compose file and use `--exit-code-from` to wait for the specific entry. Use [profiles](https://docs.docker.com/compose/profiles/) as the entry is for testing.
   It should make your workflow clean, enable running the tests easy from outside github.